xbmc/dbwrappers/dataset.cpp

```diff
@@ -321,8 +328,44 @@ bool Dataset::set_field_value(const char *f_name, const field_value &value) {
   //  return false;
 }
 
+/********* INDEXMAP SECTION START *********/
+bool Dataset::get_index_map_entry(const char *f_name) {
+  if (!(ds_state == dsEdit || ds_state == dsInsert))
+  {
+    if (~fieldIndexMapID)
+    {
+      unsigned int next(fieldIndexMapID+1 >= fieldIndexMap_Entries.size() ? 0 : fieldIndexMapID + 1);
+      if (fieldIndexMap_Entries[next].strName == f_name) //Yes, our assumption hits.
+      {
+        fieldIndexMapID = next;
+        return true;
+      }
+    }
+    // indexMap not found on the expected way, either first row strange retrival order
+    FieldIndexMapEntry tmp(f_name);
+    std::vector<unsigned int>::iterator ins(lower_bound(fieldIndexMap_Sorter.begin(), fieldIndexMap_Sorter.end(), tmp, FieldIndexMapComparator(fieldIndexMap_Entries)));
+    if (ins == fieldIndexMap_Sorter.end() || (tmp <  fieldIndexMap_Entries[*ins])) //new entry
+    {
+      //Insert the new item just behind last retrieved item
+      //In general this should be always end(), but could be different
+      fieldIndexMap_Sorter.insert(ins, ++fieldIndexMapID);
+      fieldIndexMap_Entries.insert(fieldIndexMap_Entries.begin() + fieldIndexMapID, tmp);
+    }
+    else //entry already existing!
+    {
+      fieldIndexMapID = *ins;
+      return true;
+    }
+  }
+  return false; //invalid
+}
+/********* INDEXMAP SECTION END *********/
 
 const field_value Dataset::get_field_value(const char *f_name) {
+  //Lets try to reuse a string ->index conversation
+  if (get_index_map_entry(f_name))
+    return get_field_value(static_cast<int>(fieldIndexMap_Entries[fieldIndexMapID].fieldIndex));
+
   const char* name=strstr(f_name, ".");
   if (name) name++;
   if (ds_state != dsInactive) {
```

Comment:
we should move the `name` declaration code inside this.

```
if (ds_state != dsInactive) {
  const char* name = strstr(f_name, ".");
  if (name) 
    name++;
```

and the `get_index_map_entry` stuff inside the else of `if (ds_state == dsEdit || ds_state == dsInsert){`
We could remove the check for `ds_state == dsEdit || ds_state == dsInsert` in `get_index_map_entry()` to not doing it twice.